src/wp-includes/theme.php

```diff
@@ -840,6 +840,16 @@ function switch_theme( $stylesheet ) {
 	$new_theme->delete_pattern_cache();
 	$old_theme->delete_pattern_cache();
 
+	// Set autoload=no for the old theme, autoload=yes for the switched theme.
+	$theme_mods_options = array(
+		'theme_mods_' . $stylesheet                  => 'yes',
+		'theme_mods_' . $old_theme->get_stylesheet() => 'no',
+	);
+	wp_set_option_autoload_values( $theme_mods_options );
+
+	// Reload autoload options.
+	wp_load_alloptions();
```

Comment:
I thought `wp_set_option_autoload_values` already does this, no?

Reply:
I see so, yes.

Reply:
Thanks for the feedback. I have removed the reload call in https://github.com/WordPress/wordpress-develop/pull/5706/commits/041eaad9ebae0dbeb5d5bd1af4f59b9aa19d83ac and now the unit tests going failed here: https://github.com/WordPress/wordpress-develop/actions/runs/7027995261/job/19123300587?pr=5706.

The [wp_set_option_autoload_values](https://github.com/WordPress/wordpress-develop/blob/a21f5a3cc054137f3ace4760db43845158fb3cc2/src/wp-includes/option.php#L475-L476) function remove the cache in that function and it never reload the all options cache that may be the error?

cc. @felixarntz @joemcgill

Reply:
Specifically the [error](https://github.com/WordPress/wordpress-develop/actions/runs/7027995261/job/19123300587?pr=5706#step:14:334) is:

```
1) Tests_Autoload_Theme_Mods::test_that_on_switch_theme_previous_theme_mods_should_not_be_autoload
PHPUnit\Framework\Exception: Argument #2 (No Value) of PHPUnit\Framework\Assert::assertArrayNotHasKey() must be a array or ArrayAccess
```

Failing assertion:

https://github.com/WordPress/wordpress-develop/blob/041eaad9ebae0dbeb5d5bd1af4f59b9aa19d83ac/tests/phpunit/tests/theme/autoloadThemeMods.php#L45-L46